### ISSUE TITLE
Feature request: Look for async void tests

### ISSUE THREAD

**Issue by mariusGundersen** [nunit/nunit.analyzers#7]
Since it's not allowed to have `async void` tests, there should be an analyzer that looks for these tests and autofixes them to be `async Task`.

**Comment by JasonBock** [comment]
In general an `async` method should never return `void` unless it's handling an event from the UI. So....would be nice if an analyzer was written for the general case and not just for unit tests.

**Comment by mikkelbu** [comment]
Examples 
```csharp
    [Test]
    public async void AsyncVoid()
    {
      await Task.Delay(1); // To avoid warning message
    }
```
and
```csharp
    [TestCase(4)]
    public async void AsyncVoidTestCase(int x)
    {
      await Task.Delay(0);
    }
```
These are not runnable with the following error `Message: Async test method must have non-void return type`.

**Comment by mikkelbu** [comment]
Hi @304NotModified 
When I added the milestone to this issue (and the other async-issues) this Sunday I also started working on #48 and created a new analyzer for all the async checks. 

Currently, I've only made the skeleton, as this week has been quite busy. But I think that we can move most of the logic in #69 to that class when I get it pushed - hopefully, this weekend or in the beginning of next week.
